crosslink/internal/crosslink.go

```diff
@@ -19,46 +19,68 @@ import (
 	"path/filepath"
 	"strings"
 
+	"go.uber.org/zap"
 	"golang.org/x/mod/modfile"
 )
 
-func Crosslink(rc runConfig) {
+func Crosslink(rc RunConfig) {
 	var err error
 
 	rootModulePath, err := identifyRootModule(rc.RootPath)
 	if err != nil {
-		panic(fmt.Sprintf("failed to identify root module: %v", err))
+		rc.Logger.Sugar().Panic("Failed to identify root Module",
+			zap.Error(err),
+			zap.Any("run config", rc))
 	}
 
 	graph, err := buildDepedencyGraph(rc, rootModulePath)
 	if err != nil {
-		panic(fmt.Sprintf("failed to build dependency graph: %v", err))
+		rc.Logger.Sugar().Panic("Failed to build dependency graph",
+			zap.Any("Run Config", rc),
+			zap.String("Root Module Path", rootModulePath))
 	}
 
-	for _, moduleInfo := range graph {
+	for moduleName, moduleInfo := range graph {
 		err = insertReplace(&moduleInfo, rc)
 		if err != nil {
-			panic(fmt.Sprintf("failed to insert replace statements: %v", err))
+			rc.Logger.Sugar().Error("Failed to insert replace statements",
+				zap.Error(err),
+				zap.String("Module Name", moduleName),
+				zap.Any("Module Info", moduleInfo),
```

Comment:
Is this necessary?  This can be a lot of data IIRC, maybe just include specific parts of it that are useful.